mlflow/tracking/fluent.py

```diff
@@ -400,13 +423,16 @@ def end_run(status: str = RunStatus.to_string(RunStatus.FINISHED)) -> None:
         --
         Active run: None
     """
-    global _active_run_stack, _last_active_run_id
+    global _active_run_stack, _last_active_run_id, run_id_to_system_metrics_monitor
     if len(_active_run_stack) > 0:
         # Clear out the global existing run environment variable as well.
         MLFLOW_RUN_ID.unset()
         run = _active_run_stack.pop()
-        MlflowClient().set_terminated(run.info.run_id, status)
         _last_active_run_id = run.info.run_id
+        MlflowClient().set_terminated(_last_active_run_id, status)
+        if _last_active_run_id in run_id_to_system_metrics_monitor:
+            system_metrics_monitor = run_id_to_system_metrics_monitor.pop(_last_active_run_id)
+            system_metrics_monitor.finish()
```

Comment:
```suggestion
            if monitor := run_id_to_system_metrics_monitor.pop(_last_active_run_id, None):
                monitor.finish()
```

to make the code look a bit safer.

Reply:
I missed `if _last_active_run_id in run_id_to_system_metrics_monitor:`.